transaction-overview.md

```diff
@@ -155,53 +152,20 @@ rollback;
 
-```sql
-START TRANSACTION;
-UPDATE my_table SET a = 'new_value' WHERE id = 1;
-UPDATE my_table SET a = 'newer_value' WHERE id = 2;
-UPDATE my_table SET a = 'newest_value' WHERE id = 3;
-COMMIT;
-```
+The actual individual transaction size limit also depends on the amount of remaining memory available to the server, because when a transaction is executed, the TiDB process consumes approximately six times more memory than the transaction size.
 
-Similarly, it is recommended to execute `INSERT` statements within an explicit transaction.
+Before v4.0, TiDB restricts the total number of key-value pairs for a single transaction to no more than 300,000. This limitation is removed since v4.0.
 
 > **Note:**
 >
-> The single-threaded workloads in TiDB might not fully use TiDB's distributed resources, so the performance of TiDB is lower than that of a single-instance deployment of MySQL. This difference is similar to the case of transactions with higher latency in TiDB.
-
-### Large transaction
-
-Due to the requirement of the two-phase commit, a large transaction can lead to the following issues:
-
-* OOM (Out of Memory) when excessive data is written in the memory
-* More conflicts in the prewrite phase
-* Long duration before transactions are actually committed
-
-Therefore, TiDB intentionally imposes some limits on transaction sizes:
-
-* The total number of SQL statements in a transaction is no more than 5,000 (default)
-* Each key-value pair is no more than 6 MB
-
-For each transaction, it is recommended to keep the number of SQL statements between 100 to 500 to achieve an optimal performance.
-
-TiDB sets a default limit of 100 MB for the total size of key-value pairs, which can be modified by the `txn-total-size-limit` configuration item in the configuration file. The maximum value of `txn-total-size-limit` is 10 GB. The actual size limit of one transaction also depends on the memory capacity. When executing large transactions, the memory usage of the TiDB process is approximately 6 times larger than the total size of transactions.
-
-In versions earlier than 4.0, TiDB limits the total number of key-value pairs for a single transaction to no more than 300,000. This limitation is removed since v4.0.
+> Usually, TiDB Binlog is enabled to replicate data to the downstream. In some scenarios, message middleware such as Kafka is used to consume binlogs that are replicated to the downstream.
+>
+> Taking Kafka as an example, the upper limit of Kafka's single message processing capability is 1 GB. Therefore, when `txn-total-size-limit` is set to more than 1 GB, it might happen that the transaction is successfully executed in TiDB, but the downstream Kafka reports an error. In order to avoid this situation, you need to decide the actual value of `txn-total-size-limit` according to the limit of the end consumer. For example, if Kafka is used downstream, `txn-total-size-limit` must not exceed 1 GB.
```

Comment:
```suggestion
> Taking Kafka as an example, the upper limit of Kafka's single message processing capability is 1 GB. Therefore, when `txn-total-size-limit` is set to more than 1 GB, it might happen that the transaction is successfully executed in TiDB, but the downstream Kafka reports an error. To avoid this situation, you need to decide the actual value of `txn-total-size-limit` according to the limit of the end consumer. For example, if Kafka is used downstream, `txn-total-size-limit` must not exceed 1 GB.
```